operators/src/engine/query_processor.rs

```diff
@@ -39,11 +44,15 @@ pub trait QueryProcessor: Send + Sync {
         query: QueryRectangle<Self::SpatialBounds, Self::Selection>, // TODO: query by reference
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<Self::Output>>> {
+        self.result_descriptor().validate_query(&query)?;
```

Comment:
this will validate each operator for each query, right? can it be too expensive?

Reply:
At every query processors something can got wrong. It would maybe make sense to only check it at the root operator, and make it a debug assert for all other operators, because then it most likely is a bug, if an operator queries its source with a non-existing band.

Then again the bands may also depend on an operators parameters.

Also I am not sure if we even know if a query processor is at the root level. We only know that for operators and not at query time, right? To get this information there  would most likely require another 1000 lines changed.